ui/src/app/components/app-router.tsx

```diff
@@ -82,8 +85,31 @@ export const withFullHeight = WrappedComponent => ({...props}) => {
   return <div style={{height: '100%'}}><WrappedComponent {...props} /></div>;
 };
 
+const getUserConfirmation = (message, callback) => {
```

Comment:
I gather that this is supposed to show up when we display the prompt but I don't understand how they talk to each other. Could you explain how this works?

Reply:
yeah. I don't know exactly how the implementation works but usually, adding `<Prompt>` to a component will trigger the "Are you sure" browser modal to pop up when it unmounts. There's an alternative behavior you can specify which is providing it a `getUserConfirmation` callback function which will be invoked instead of popping up the browser's warning modal.

`message` is what's provided through the `<Prompt>` prop whereas `callback` is a function we need to call to indicate that the user wants to stay or leave.

Since this `getUserConfirmation` function just executes (doesn't render anything inherently), we need to add the rendering code ourselves which is what the `ReactDOM` bit is doing.

Reply:
Gotcha, thank you! This explains why it is a prop on the `BrowserRouter` as well.